src/controllers/issue-controller.ts

```diff
@@ -36,6 +37,9 @@ export class IssueController {
 
             this.redmine.addTimeEntry(this.issue.id, activity.id, hours, message).then(() => {
                 vscode.window.showInformationMessage(`Time entry for issue #${this.issue.id} has been added.`);
+            },(err)=>{
+                vscode.window.showWarningMessage(err)
+                .then(() => this.setTimeEntryMessage(activity), () => this.setTimeEntryMessage(activity));
```

Comment:
This might result into infinite loop, if redmine server will keep sending us response code >=400